openapi/models/smart-contract-write-params.yaml

```diff
@@ -4,11 +4,45 @@ properties:
     type: string
     description: The address of the smart contract
   abi:
-    type: string
-    description: The ABI of the smart contract
+    description: The JSON ABI of the smart contract (contains only required details. more details here https://docs.soliditylang.org/en/v0.8.19/abi-spec.html#json)
+    type: object
+    properties:
+      type:
+        type: string
+        enum:
+          - function
+      name:
+        type: string
+        example: approve
+        description: The name of the function
+      inputs:
+        type: array
+        items:
+          type: object
+          properties:
+            name:
+              description: The name of the parameter
+              type: string
+              example: _spender
+            type:
+              description: The type of the parameter
+              type: string
+              example: address
+      outputs:
+        type: array
+        items:
+          type: object
+          properties:
+            name:
+              description: The name of the parameter
+              type: string
+            type:
+              description: The type of the parameter
+              type: string
+              example: uint256
   method:
     type: string
     description: Smart contract method
   params:
     type: object
-    description: Smart contract call params
+    description: Smart contract call params. Keys are matching input names defined in ABI
```

Comment:
Keys are matching names and order defined in ABI

Reply:
ty. updated.